controlplane/internal/controller/openshiftassistedcontrolplane_controller_test.go

```diff
@@ -273,6 +274,123 @@ var _ = Describe("OpenshiftAssistedControlPlane Controller", func() {
+				)
+
+				By("confirming an error is returned when calling getWorkloadClusterVersion")
+				_, err := controllerReconciler.getWorkloadClusterVersion(ctx, openshiftAssistedControlPlane)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(Equal("kubeconfig for workload cluster is not available yet"))
+			})
+		})
+
+		When("the kubeconfig secret is available", func() {
+			It("should return the openshiftassistedcontrolplane's distributionversion", func() {
+				By("creating the openshiftassistedcontrolplane with condition kubeconfig available set to true")
+				openshiftAssistedControlPlane := getOpenshiftAssistedControlPlane()
+				openshiftAssistedControlPlane.Labels = map[string]string{
+					clusterv1.ClusterNameLabel: cluster.Name,
+				}
+				conditions.MarkTrue(openshiftAssistedControlPlane, controlplanev1alpha2.KubeconfigAvailableCondition)
+				By("creating the kubeconfig secret")
+				kubeconfigSecret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-kubeconfig", cluster.Name),
+						Namespace: openshiftAssistedControlPlane.Namespace,
+					},
+					Data: map[string][]byte{
+						"value": []byte("kubeconfig"),
+					},
+				}
+				Expect(k8sClient.Create(ctx, kubeconfigSecret)).To(Succeed())
+				By("creating a cluster version using the workload cluster's client")
+				mockWorkloadClientGenerator.MockCreateClusterVersion(openshiftAssistedControlPlane.Spec.DistributionVersion)
+				By("checking the cluster version returned")
+				distributionVersion, err := controllerReconciler.getWorkloadClusterVersion(ctx, openshiftAssistedControlPlane)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(distributionVersion).To(Equal(openshiftAssistedControlPlane.Spec.DistributionVersion))
+			})
+		})
+	})
+	Context("isUpgradeRequested", func() {
+		When("openshiftassistedcontrolplane doesn't have the distributionVersion status set", func() {
+			It("returns false", func() {
+				openshiftAssistedControlPlane := getOpenshiftAssistedControlPlane()
+				Expect(isUpgradeRequested(context.Background(), openshiftAssistedControlPlane)).To(BeFalse())
+			})
+		})
+		When("openshiftassistedcontrolplane's requested upgrade is less than the current workload cluster's version", func() {
+			It("returns false", func() {
+				openshiftAssistedControlPlane := getOpenshiftAssistedControlPlane()
+				openshiftAssistedControlPlane.Status.DistributionVersion = "4.18.0"
+				Expect(isUpgradeRequested(context.Background(), openshiftAssistedControlPlane)).To(BeFalse())
+			})
+		})
+		When("openshiftassistedcontrolplane's requested upgrade is greater than the current workload cluster's version", func() {
+			It("returns true", func() {
+				openshiftAssistedControlPlane := getOpenshiftAssistedControlPlane()
+				openshiftAssistedControlPlane.Status.DistributionVersion = "4.11.0"
+				Expect(isUpgradeRequested(context.Background(), openshiftAssistedControlPlane)).To(BeTrue())
```

Comment:
Same here: we are testing private methods.

How would the upgrade look like? If there is an updater or something, we could mock and assert it has been called. In case this approach won't work, we can leave the test for now and substitute it when we'll integrate the actual logic of upgrades (and make some assertion on it). WDYT?